### ISSUE TITLE
GNOME Shell collapses when using dfd extension.

### ISSUE THREAD

**Issue by Betha0d** [fthx/dock-from-dash#95]
I 'm using Fedora 39/GNOME 45/Wayland. When I turn on dfd (only), close the extension setting window, open GNU Octave, close GNU Octave, the screen just freezes and then GNOME Shell collapses.
 The collapse appears every time when I try the action above, but it doesn't seem to happen to other apps on my computer. However,  I have no idea what log (and other info) I should provide.

**Comment by fthx** [comment]
So you confirm it does not happen without any extension activated and it does happen with DfD as the only extension activated?

If true, could you search for "js" in Logs (all messages)?

**Comment by Betha0d** [comment]
> So you confirm it does not happen without any extension activated and it does happen with DfD as the only extension activated?
> 
> If true, could you search for "js" in Logs (all messages)?

Yes. And I forget to mention that the collapse only happens when a menu on the top of the window of Octave is open.
![截图 2023-11-29 22-59-04](https://github.com/fthx/dock-from-dash/assets/78027552/20a4be4b-4fda-49f7-acea-51ed3b56ca37)
This is what I get when I search for"js" in Logs. Hope this may help.

**Comment by fthx** [comment]
I installed Octave and I don't hit this at all.
I see that you have another extension (gamemode) running?

**Comment by Betha0d** [comment]
> I installed Octave and I don't hit this at all. I see that you have another extension (gamemode) running?

Gamemode was installed by default, and I didn't enable it. There is an error with it (SyntaxError: import declarations may only appear at top level of a module). I restarted my computer, created a new user, and found that the collapse can still happen sometimes even though no new extensions are installed and all default extensions are disabled. (It happened every time before I restarted, while enabling dfd). 
So I suggest this collapse has nothing to do with dfd. Maybe this issue can be closed now. Thx for your patience.
